index.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const cdsServices = require('./discovery/cds-services');
+
+const app = express();
+
+const isValidJson = (obj) => {
+  try {
+    JSON.stringify(obj);
+    return true;
+  } catch (err) {
+    return false;
+  }
+};
+
+// This is necessary middleware to parse JSON into the incoming request body for POST requests
+app.use(bodyParser.json());
+
+// CDS Services must implement CORS to be called from a web browser
+app.use((request, response, next) => {
+  response.set({
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Methods': 'GET, POST, OPTIONS',
+    'Access-Control-Allow-Credentials': 'true',
+    'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+    'Access-Control-Expose-Headers': 'Origin, Accept, Content-Location, Location, X-Requested-With',
+    'Content-Type': 'application/json; charset=utf-8',
+  });
+  next();
+});
+
+app.use((request, response, next) => {
+  if (request.body) {
+    if (!isValidJson(request.body)) {
+      response.sendStatus(400);
+      return;
+    }
+  }
+  next();
+});
+
+
+app.use('/cds-services', cdsServices);
+
```

Comment:
You should probably add code to catch unhanded routes and errors. Something like this.
```  // catch 404 and forward to error handler
  app.use((req, res, next) => {
    const err = new Error('Not Found');
    err.status = 404;
    next(err);
  });

  // error handler
  app.use((err, req, res) => {
    // set locals, only providing error in development
    res.locals.message = err.message;
    res.locals.error = req.app.get('env') === 'development' ? err : {};

    // render the error page
    res.status(err.status || 500);
    res.render('error');
  });
```

Reply:
👍

Reply:
Addressed in https://github.com/cds-hooks/sandbox-cds-services/pull/1/commits/5611a7fbba491d40267a41e095c0585c566fc7da